### PR TITLE
Add docker-in-docker Azure DNS override flag

### PR DESCRIPTION
**Pull request by kklopfenstein**
By default the docker-in-docker script passes an Azure specific IP address to dockerd that causes Docker not to use the devcontainer's resolv.conf settings. We would like to not override DNS within Docker in some situations.

This PR adds a new `AZURE_DNS_OVERRIDE` flag that will give us the ability to disable this Azure DNS setting.

### CODE REVIEW

**Review by CleanCut — APPROVED** [review]

File: script-library/docker-in-docker-debian.sh [hunk 1]
```diff
@@ -7,13 +7,14 @@
 # Docs: https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/docker-in-docker.md
 # Maintainer: The VS Code and Codespaces Teams
 #
-# Syntax: ./docker-in-docker-debian.sh [enable non-root docker access flag] [non-root user] [use moby] [Engine/CLI Version] [Major version for docker-compose]
+# Syntax: ./docker-in-docker-debian.sh [enable non-root docker access flag] [non-root user] [use moby] [Engine/CLI Version] [Major version for docker-compose] [azure DNS override]
```
Comment (samruddhikhandale) [inline script-library/docker-in-docker-debian.sh:10]: Hi, thanks for the PR. 

However, we have migrated the Features to the [devcontainers/features](https://github.com/devcontainers/) repo. More details in the [announcement issue.](https://github.com/microsoft/vscode-dev-containers/issues/1589)

Do you mind opening the PR in [docker-in-docker](https://github.com/devcontainers/features/tree/main/src/docker-in-docker) Feature instead?

File: script-library/docker-in-docker-debian.sh [hunk 2]
```diff
@@ -325,6 +326,10 @@ tee /usr/local/share/docker-init.sh > /dev/null \
 
 set -e
 
+AZURE_DNS_OVERRIDE=$AZURE_DNS_OVERRIDE
+EOF
+
+tee -a ./docker-init.sh > /dev/null << 'EOF'
```
Comment (joshspicer) [inline script-library/docker-in-docker-debian.sh:332]: nit: Could we keep the style of this `tee` command the same as the one above (same line breaks, using the full `/usr/local/share/docker-init.sh` file path)

**Review by joshspicer — COMMENTED** [review]
Thank you for improving our scripts! 

As @samruddhikhandale mentioned, it would be great to additionally put up a PR on the [associated Feature](https://github.com/devcontainers/features/tree/main/src/docker-in-docker), as we have deprecated these scripts in favor of the new Features repo.  The script in the Features repo should be nearly identical.

In the Features repo, you'd want to update both the [install script](https://github.com/devcontainers/features/tree/main/src/docker-in-docker) and add in a new [`AzureDnsOverride` option](https://github.com/devcontainers/features/blob/main/src/docker-in-docker/devcontainer-feature.json#L7) so that we can surface this flag in tooling

File: script-library/docker-in-docker-debian.sh (hunk 1 above)
Comment (kklopfenstein) [reply script-library/docker-in-docker-debian.sh:10]: Here's the features PR: https://github.com/devcontainers/features/pull/200

As discussed, we would like to make this change here first and follow up with the features change.

**Pull request merged**